agent/engine/containermetadata/metadata_manager_windows.go

```diff
@@ -0,0 +1,160 @@
+// +build windows
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	"github.com/cihub/seelog"
+)
+
+const (
+	inspectContainerTimeout = 30 * time.Second
+	mountPoint              = `C:\ecs\metadata`
+)
+
+// MetadataManager is an interface that allows us to abstract away the metadata
+// operations
+type MetadataManager interface {
+	CreateMetadata([]string, *api.Task, *api.Container) ([]string, error)
+	UpdateMetadata(string, *api.Task, *api.Container) error
+	CleanTaskMetadata(*api.Task) error
+}
+
+// metadataManager implements the MetadataManager interface
+type metadataManager struct {
+	client dockerDummyClient
+	cfg    *config.Config
+}
+
+// NewMetadataManager creates a metadataManager for a given DockerTaskEngine settings.
+func NewMetadataManager(client dockerDummyClient, cfg *config.Config) MetadataManager {
+	seelog.Debug("Creating metadata manager")
+	return &metadataManager{
+		client: client,
+		cfg:    cfg,
+	}
+}
+
+// CreateMetadata creates the metadata file and adds the metadata directory to
+// the container's mounted host volumes
+// binds []string is passed by value to avoid race conditions by multiple
+// calls to CreateMetadata, although this should never actually happen
+func (manager *metadataManager) CreateMetadata(binds []string, task *api.Task, container *api.Container) ([]string, error) {
+	// Check if manager has invalid entries
+	var err error
+	if manager.cfg == nil {
+		err = fmt.Errorf("Invalid configuration")
+		return binds, err
+	}
+
+	// Do not create metadata file for internal containers
+	// Add error handling for this case? Probably no need since
+	// Internal containers should not be visible to users anyways
+	if container.IsInternal {
+		return binds, nil
+	}
+
+	// Create task and container directories if they do not yet exist
+	mdDirectoryPath, err := getMetadataFilePath(task, container, manager.cfg.DataDir)
+	// Stop metadata creation if path is malformed for any reason
+	if err != nil {
+		err = fmt.Errorf("Invalid metadata file path due to malformed input")
```

Comment:
Why not include the `err` in the error message? Also these two lines can be combined.